modules/hostap/src/supp_api.c

```diff
@@ -326,6 +336,28 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
 	}
 
 	if (params->security != WIFI_SECURITY_TYPE_NONE) {
+
+		if (params->sae_password) {
+			if ((params->sae_password_length < WIFI_PSK_MIN_LEN) ||
+			    (params->sae_password_length > WIFI_SAE_PSWD_MAX_LEN)) {
+				wpa_printf(MSG_ERROR,
+					   "Passphrase should be in range (%d-%d) characters",
+					   WIFI_PSK_MIN_LEN, WIFI_SAE_PSWD_MAX_LEN);
+				goto out;
+			}
+		} else {
+			if ((params->psk_length < WIFI_PSK_MIN_LEN) ||
+			    (params->psk_length > WIFI_PSK_MAX_LEN)) {
+				wpa_printf(MSG_ERROR,
+					   "Passphrase should be in range (%d-%d) characters",
+					   WIFI_PSK_MIN_LEN, WIFI_PSK_MAX_LEN);
+				goto out;
+			}
+		}
+
+		strncpy(psk_null_terminated, params->psk, WIFI_PSK_MAX_LEN);
+		psk_null_terminated[params->psk_length] = '\0';
```

Comment:
Shouldn't these use `params->sae_password` and `params->sae_password_length` for SAE?

Reply:
Yes it should be. Currently there is one crash because of this `strncpy`. Debugging the issue.